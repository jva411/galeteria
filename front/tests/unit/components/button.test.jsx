/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../../../components/button'
import userEvent from '@testing-library/user-event'
import { render, screen } from '@testing-library/react'

describe('unit test of component \'button\'', () => {
    const child = 'button-text'

    test('should receive type passed', () => {
        render(<Button type='submit'>{child}</Button>)
        
        expect(screen.getByRole('button')).toHaveProperty('type', 'submit')
    })

    test('should call onClick on click event', () => {
        const handleClick = jest.fn()

        render(<Button onClick={handleClick}>{child}</Button>)
        
        userEvent.click(screen.getByRole('button'))

        expect(handleClick).toHaveBeenCalledTimes(1)
    })

    test('should Button match with Snapshot', () => {
        const tree = renderer.create(<Button type='submit'>{child}</Button>).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
