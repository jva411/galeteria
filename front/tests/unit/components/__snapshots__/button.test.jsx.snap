// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test of component 'button' should Button match with Snapshot 1`] = `
<button
  type="submit"
>
  button-text
</button>
`;
